BossMod/Components/RoleplayModule.cs

```diff
@@ -0,0 +1,44 @@
+﻿namespace BossMod.Components;
+public abstract class RoleplayModule(BossModule module) : BossComponent(module)
+{
+    private AIHints? _hints;
+    private Actor? _player;
+    protected AIHints Hints => _hints!;
+    protected Actor Player => _player!;
+
+    protected Roleplay.AID ComboAction => (Roleplay.AID)WorldState.Client.ComboState.Action;
+
+    public abstract void Execute(Actor? primaryTarget);
+
+    public override void AddAIHints(int slot, Actor actor, PartyRolesConfig.Assignment assignment, AIHints hints)
+    {
+        _hints = hints;
+        _player = actor;
+
+        Execute(WorldState.Actors.Find(actor.TargetID) ?? Module.PrimaryActor);
+    }
+
+    protected void UseGCD(Roleplay.AID action, Actor? target, Vector3 targetPos = default)
+        => Hints.ActionsToExecute.Push(ActionID.MakeSpell(action), target, ActionQueue.Priority.High, targetPos: targetPos);
+
+    protected void UseOGCD(Roleplay.AID action, Actor? target, Vector3 targetPos = default)
+        => Hints.ActionsToExecute.Push(ActionID.MakeSpell(action), target, ActionQueue.Priority.Low, targetPos: targetPos);
```

Comment:
A lot of this makes sense as utilities for rotation modules, and I appreciate that 'roleplay components' are kind of similar, but does it really make too much sense? Not like anyone is going to create a cdplan for rp battles, so eg priorities don't really need to adhere to the actionqueue conventions; don't think gcd/ogcd distinction is even too meaningful during rp battles...

Reply:
the hien battle against zenos at the end of stb actually does have gcd vs ogcd distinction., not that it really matters, since that whole battle is a joke - the point is to stall against wol - but it is at least nice to have a function defined for it. i can remove these and just replace them with one function called `UseAction` for example if that would be better